src/MongoDB.Driver.Core/Core/Authentication/ScramShaAuthenticator.cs

```diff
@@ -138,9 +138,6 @@ private class ScramShaMechanism : ISaslMechanism
 
             public ISaslStep Initialize(IConnection connection, SaslConversation conversation, ConnectionDescription description)
             {
-                Ensure.IsNotNull(connection, nameof(connection));
```

Comment:
We need to remove this check because when we speculatively authenticate, we don't yet have a connection, but we to create this `ISaslStep` to include in the initial `isMaster`